components/ParsePOSPayment/Receive-Invoice.tsx

```diff
@@ -0,0 +1,159 @@
+import copy from "copy-to-clipboard"
+import { useRouter } from "next/router"
+import React from "react"
+import Image from "react-bootstrap/Image"
+import OverlayTrigger from "react-bootstrap/OverlayTrigger"
+import Tooltip from "react-bootstrap/Tooltip"
+import { QRCode } from "react-qrcode-logo"
+import { useTimer } from "react-timer-hook"
+
+import useCreateInvoice from "../../hooks/use-Create-Invoice"
+import { LnInvoiceObject } from "../../lib/graphql/index.types.d"
+import useSatPrice from "../../lib/use-sat-price"
+import { ACTION_TYPE } from "../../pages/merchant/_reducer"
+import { formatOperand } from "../../utils/utils"
+import PaymentOutcome from "../PaymentOutcome"
+import styles from "./parse-payment.module.css"
+
+interface Props {
+  recipientWalletCurrency?: string
+  walletId: string | undefined
+  state: React.ComponentState
+  dispatch: React.Dispatch<ACTION_TYPE>
+}
+
+function ReceiveInvoice({ recipientWalletCurrency, walletId, dispatch }: Props) {
+  const { usdToSats } = useSatPrice()
+  const { amount, currency } = useRouter().query
+  const [copied, setCopied] = React.useState<boolean>(false)
+
+  const { createInvoice, data, error, loading } = useCreateInvoice({
+    recipientWalletCurrency,
+  })
+  const paymentAmount = React.useMemo(() => {
+    if (currency === "USD") {
+      return amount
+    }
+    return formatOperand(usdToSats(Number(amount)).toFixed().toString())
+  }, [amount, currency, usdToSats])
+
+  const generateInvoice = React.useCallback(() => {
+    createInvoice({
+      variables: { walletId: walletId, amount: paymentAmount },
+    })
+  }, [paymentAmount, walletId, createInvoice])
+
+  React.useEffect(() => {
+    generateInvoice()
+  }, [generateInvoice])
+
+  let errorString: string | null = error?.message || null
+  let invoice: LnInvoiceObject | undefined
+
+  if (data) {
+    const invoiceData = data.mutationData
+    if (invoiceData.errors?.length > 0) {
+      errorString = invoiceData.errors.map((e) => e.message).join(", ")
+    } else {
+      invoice = invoiceData.invoice
+    }
+  }
+
+  const loadingOrErrorMsg =
+    loading || !invoice?.paymentRequest ? (
+      <p className={styles.loading}>Generating invoice</p>
+    ) : error || data?.mutationData.errors.length ? (
+      <p className={styles.error}>{errorString}</p>
+    ) : null
+
+  const GetTimer = () => {
+    const time = new Date()
+    time.setSeconds(time.getSeconds() + 60 * 5) // default to five mins for USD invoice
+    const expiryTimestamp = time
+    const { seconds, minutes } = useTimer({
```

Comment:
Hooks should only be used at the top level or in other hooks, so this should be refactored so that ```useTimer``` in line 73 is outside of the GetTimer function. https://reactjs.org/docs/hooks-rules.html

Reply:
Done!